app/v1/controllers/CreateAndAmendCharitableGivingController.scala

```diff
@@ -0,0 +1,124 @@
+      val result =
+        for {
+          parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
+          serviceResponse <- EitherT(service.amend(parsedRequest))
+          vendorResponse <- EitherT.fromEither[Future](
+            hateoasFactory
+              .wrap(serviceResponse.responseData, CreateAndAmendCharitableGivingTaxReliefHateoasData(nino, taxYear))
+              .asRight[ErrorWrapper])
+        } yield {
+          logger.info(
+            s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
+              s"Success response received with CorrelationId: ${serviceResponse.correlationId}")
+
+          val response = Json.toJson(vendorResponse)
+
+//          auditSubmission(
+//            CreateAndAmendCharitableGivingAuditDetail(
+//              request.userDetails,
+//              nino,
+//              taxYear,
+//              request.body,
+//              serviceResponse.correlationId,
+//              AuditResponse(OK, Right(Some(response)))))
+
+          Ok(response)
+            .withApiHeaders(serviceResponse.correlationId)
+        }
+
+      result.leftMap { errorWrapper =>
+        val resCorrelationId = errorWrapper.correlationId
+        val result           = errorResult(errorWrapper).withApiHeaders(resCorrelationId)
+
+        logger.warn(
+          s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
+            s"Error response received with CorrelationId: $resCorrelationId")
+
+//        auditSubmission(
+//          CreateAndAmendCharitableGivingAuditDetail(
+//            request.userDetails,
+//            nino,
+//            taxYear,
+//            request.body,
+//            correlationId,
+//            AuditResponse(result.header.status, Left(errorWrapper.auditErrors))))
+
+        result
+      }.merge
+    }
+
+  private def errorResult(errorWrapper: ErrorWrapper) = {
+
+    errorWrapper.error match {
+      case NinoFormatError | BadRequestError | TaxYearFormatError | RuleTaxYearNotSupportedError | RuleTaxYearRangeInvalidError |
+          RuleGiftAidNonUkAmountWithoutNamesError | RuleGiftsNonUkAmountWithoutNamesError | MtdErrorWithCustomMessage(StringFormatError.code) |
+          MtdErrorWithCustomMessage(ValueFormatError.code) | MtdErrorWithCustomMessage(RuleIncorrectOrEmptyBodyError.code) =>
+        BadRequest(Json.toJson(errorWrapper))
+      case NotFoundError   => NotFound(Json.toJson(errorWrapper))
+      case DownstreamError => InternalServerError(Json.toJson(errorWrapper))
+      case _               => unhandledError(errorWrapper)
+    }
+  }
+
+//  private def auditSubmission(details: CreateAndAmendCharitableGivingAuditDetail)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+//    val event = AuditEvent("CreateAmendCharitableGivingReliefs", "create-amend-charitable-giving-reliefs", details)
+//    auditService.auditEvent(event)
```

Comment:
Probably best to remove the commented code from the controller since it'll be awhile before we do audits